forbidigo/forbidigo.go

```diff
@@ -164,14 +168,54 @@ func (v *visitor) Visit(node ast.Node) ast.Visitor {
 		}
 		return v
 	case *ast.SelectorExpr:
+		selectorExpr = node
 	case *ast.Ident:
 	default:
 		return v
 	}
+
+	// The text as it appears in the source is always used because issues
+	// use that. The other texts to match against are extracted when needed
+	// by a pattern.
+	srcText := v.textFor(node)
+	pkgText := ""
 	for _, p := range v.linter.patterns {
-		if p.pattern.MatchString(v.textFor(node)) && !v.permit(node) {
+		if p.matchPackage && pkgText == "" {
+			if selectorExpr == nil {
```

Comment:
Minor point but this part could be pulled out into a helper method to make the intent of this method more readable.  I think I did ask for it to be inlined but I believe that was when this code compared the compare, rather than just having a function that generates the text with the absolute package name.

Reply:
Never mind, I understand what was meant now. I'm changing it...